### PR TITLE
[16.0][FIX] account_asset_management: create refund invoices with assets

### PR DESCRIPTION
**Pull request by JavierIniesta**
Previous to this fix, the functionality was working when an invoice was posted when called from action_post. But in cases as reversals action_post is not used for posting the reversal invoice, _post function is used directly.

To solve the issue, the functionality is moved from action_post to _post, to ensure it's being executed in all invoice post scenarios.

### CODE REVIEW

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
@OCA/accounting-maintainers someone to review the PR? Thanks

**Review by MayteGLC — APPROVED** [review]
LGTM

File: account_asset_management/models/account_move.py [hunk 1]
```diff
@@ -84,8 +84,8 @@ def _prepare_asset_vals(self, aml):
             "date_start": self.date,
         }
 
-    def action_post(self):
-        ret_val = super().action_post()
+    def _post(self, soft=True):
```
Comment (pedrobaeza) [inline account_asset_management/models/account_move.py:87]: I think this is not related with the title of the PR.

Comment (aritzolea) [reply account_asset_management/models/account_move.py:87]: The problem is the functionality was working when an invoice was posted when called from `action_post`. Put in cases as reversals `action_post` is not used for posting the reversal invoice, `_post` function is used directly.

That's why the funcionality is moved from `action_post` to `_post`, to ensure it's being executed in all invoice post scenarios.

Comment (pedrobaeza) [reply account_asset_management/models/account_move.py:87]: Then you should explain all of this in the commit message and in the initial comment of the PR.

Comment (aritzolea) [reply account_asset_management/models/account_move.py:87]: Added an explanation on the commit message. I cannot modify the initial comment of the PR unless I close this one and create another PR, the author of the PR is no longer working with us.

**Review by pedrobaeza — APPROVED** [review]
OK, no problem. I have copied the commit message to the initial PR message.

But please fw-port this to v17.

/ocabot merge patch
